### DISCUSSION TITLE
Renaming Organizations & Groups Thru all CKAN

### DISCUSSION THREAD

**Discussion by notboringstats** [ckan/ckan#6514] [Q&A]
As in the title how can i manage  by using an extension to change the words "Organization" and "Groups" thru all my CKAN website ? I tried several times failing miserably, can someone ELI5 ?

**Comment by frafra** [comment]
I just customize the language files, actually, using a mass find and replace.

**Comment by jqnatividad** [comment]
@notboringstats I had to do this recently, and as @frafra said, you need to customize the language files.

For more explicit step-by-step guidance, I basically followed the procedure here - https://docs.ckan.org/en/2.9/contributing/i18n.html#manual-setup

Assuming you're using the en locale, on step 2,  that means modifying the en.po file and mapping "Organization"/s and "Group"/s (because you need to explicit map the plural form as well).

**Comment by smotornyuk** [comment]
[Here you can find basic extension](https://github.com/smotornyuk/ckanext-example-organization-label) that translates more or less all the `organization`'s to `team`'s.

in order to test it, download the source, run `pip install -e path/to/downloaded/repo` and add `example_organization_label` to `ckan.plugins` config option


What I've done:

* located all the records(msgid/msgstr pair) with `organization` inside the CKAN [translation vocabulary](https://github.com/ckan/ckan/blob/master/ckan/i18n/ckan.pot)
* [Copied them to my extension](https://github.com/smotornyuk/ckanext-example-organization-label/blob/master/ckanext/example_organization_label/i18n/ckanext-example_organization_label.pot) and provided "translation" to the expected term
* Created and compiled "improved" version of labels via two following commands: 
  ```
  python setup.py init_catalog -l en
  python setup.py compile_catalog -l en
  ``` 


You can use this extension as a base and adapt it to your needs. In order to translate `groups`:

* search for all the lines that contain `group` inside CKAN [translation source](https://github.com/ckan/ckan/blob/master/ckan/i18n/ckan.pot)
* [add them to your extension extension](https://github.com/smotornyuk/ckanext-example-organization-label/blob/master/ckanext/example_organization_label/i18n/ckanext-example_organization_label.pot)
* change `msgstr` to the value you want to see(do not change `msgid`)
* Compile translations:
  ```
  python setup.py init_catalog -l en
  python setup.py compile_catalog -l en
  ``` 


You may still observe some `organization` labels(in URL, for example). In order to get rid of them, you can create *custom organization type* via [ckanext-scheming](https://github.com/ckan/ckanext-scheming#example-schemas---organization). 

in the next CKAN release, there will be a simpler way to achieve what you want. Check #5337, and consider whether it is worth waiting for the next CKAN release instead of implementing it right now
